spatialmath/geo_obstacle.go

```diff
@@ -205,52 +197,44 @@ func (gpo *GeoPose) Heading() float64 {
 	return gpo.heading
 }
 
-// PoseToGeoPose converts a pose (in MM) into a GeoPose treating relativeTo as the origin.
-func PoseToGeoPose(relativeTo GeoPose, pMM Pose) GeoPose {
-	// we do this b/c plan nodes describe movement in mm
-	// but (p *Point) PointAtDistanceAndBearing expects the pose to be in km
-	kmPoint := r3.Vector{
-		X: pMM.Point().X / 1e6,
-		Y: pMM.Point().Y / 1e6,
-		Z: pMM.Point().Z / 1e6,
-	}
-	p := NewPose(kmPoint, pMM.Orientation())
-
-	// math.Atan2 performs on the unit sphere which is right-handed
-	// we assign newX and newY so we are peforming a left-handed calculation
-	newX := -p.Point().X
-	newY := p.Point().Y
-	bearingRad := math.Atan2(newX, newY)
-
-	// convert bearingRad to degrees
-	bearingDeg := bearingRad * 180 / math.Pi * -1
-
-	// get the maginitude of pose p
-	poseMagnitude := p.Point().Norm()
-
-	// relativeTo Heading is a right-handed value
+// PoseToGeoPose converts a pose (which are always in mm) into a GeoPose treating relativeTo as the origin.
+func PoseToGeoPose(relativeTo *GeoPose, pose Pose) *GeoPose {
+	// poses are always in mm but PointAtDistanceAndBearing expects the pose to be in km so we need to convert
+	mmPoint := pose.Point()
+	kmPoint := r3.Vector{X: mmPoint.X / 1e6, Y: mmPoint.Y / 1e6, Z: mmPoint.Z / 1e6}
+
+	// calculate the bearing (illustrated on the plot below as angle "x"), to the GeoPose (illustrated as "*")
+	//       |   *
+	//       |x /
+	//       | /
+	//       |/
+	// -----------
+	//       |
+	//       |
+	bearing := utils.RadToDeg(math.Atan2(kmPoint.X, kmPoint.Y))
```

Comment:
I take it this returns a left-handed value?

Reply:
yep

Reply:
yes, as we are measuring the angle from the right side of the y axis it is left handed.  will add a comment.